concept/type_/relation_type.rs

```diff
@@ -626,6 +626,22 @@ impl RelationType {
         }
         Ok(())
     }
+
+    pub fn relation_index_available<Snapshot: ReadableSnapshot>(
+        &self,
+        snapshot: &Snapshot,
+        type_manager: &TypeManager,
+    ) -> Result<bool, Box<ConceptReadError>> {
+        Ok(Snapshot::STATIC_SCHEMA_GUARANTEE && self.schema_qualifies_for_relation_index(snapshot, type_manager)?)
```

Comment:
Err let's call it `IMMUTABLE_SCHEMA`

Reply:
if you're introducing a "new" term ('static') always have a think if we don't already have one!